### PR TITLE
Test: unit test for features controller and services

### PR DESCRIPTION
**Pull request by Ajen07**
# Description

- Unit tests for `Feature controller and Services`

## Issue link

[Fixes # (issue)](https://app.clickup.com/t/86azu6kw4)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- tested with  `yarn test:docker`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: src/features/features.service.spec.ts [hunk 1]
```diff
@@ -12,16 +112,799 @@ describe("FeaturesService", () => {
...
+            expect(service.updateFeatureOrderAndCategory).toBeDefined();
+        });
+
+        it("should update a feature to different feature category", async () => {
+            const FeaturesArray = [
+                {
+                    id: 1,
+                    teamMemberId: 1,
+                    createdAt: new Date(Date.now()),
+                    updatedAt: new Date(Date.now()),
```
Comment (timDeHof) [inline src/features/features.service.spec.ts:476]: I think we should use a constant variable that holds a specific date and time.This way we can ensure consistent results regardless of the current date, time, or timezone.
```suggestion
                    // maybe written at the top in the describe
                    const mockDate = new Date("2024-11-05T02:41:03.575Z");
                    createdAt: mockDate,
                    updatedAt: mockDate,
```

**Review by timDeHof — CHANGES_REQUESTED** [review]
All tests passed. But I added a suggestion for the dates for `createdAt` and `updatedAt`

**Review by cherylli — APPROVED** [review]
thanks. went through the code and ran all the tests, no issue.

**Review by timDeHof — APPROVED** [review]
Everything looks great. Thank you for making that change.

**Pull request merged**